src/triage/component/catwalk/evaluation.py

```diff
@@ -211,65 +317,14 @@ def evaluate(
             labels,
             self.sort_seed
         )
-        labels_sorted = numpy.array(labels_sorted)
 
         evaluations = []
         for group in self.metric_groups:
-            logging.info('Creating evaluations for metric group %s', group)
-            parameters = group.get('parameters', [{}])
-            if 'thresholds' not in group:
-                logging.info('Not a thresholded group, generating evaluation '
-                             'based on all predictions')
-                evaluations = evaluations + self._generate_evaluations(
-                    group['metrics'],
-                    parameters,
-                    {},
-                    predictions_proba,
-                    generate_binary_at_x(
-                        predictions_proba_sorted,
-                        100,
-                        unit='percentile'
-                    ),
-                    labels_sorted.tolist(),
-                )
-
-            for pct_thresh in group.get('thresholds', {}).get('percentiles', []):
-                logging.info('Processing percent threshold %s', pct_thresh)
-                predicted_classes = numpy.array(generate_binary_at_x(
-                    predictions_proba_sorted,
-                    pct_thresh,
-                    unit='percentile'
-                ))
-                nan_mask = numpy.isfinite(labels_sorted)
-                predicted_classes = (predicted_classes[nan_mask]).tolist()
-                present_labels_sorted = (labels_sorted[nan_mask]).tolist()
-                evaluations = evaluations + self._generate_evaluations(
-                    group['metrics'],
-                    parameters,
-                    {'pct': pct_thresh},
-                    None,
-                    predicted_classes,
-                    present_labels_sorted,
-                )
-
-            for abs_thresh in group.get('thresholds', {}).get('top_n', []):
-                logging.info('Processing absolute threshold %s', abs_thresh)
-                predicted_classes = numpy.array(generate_binary_at_x(
-                    predictions_proba_sorted,
-                    abs_thresh,
-                    unit='top_n'
-                ))
-                nan_mask = numpy.isfinite(labels_sorted)
-                predicted_classes = (predicted_classes[nan_mask]).tolist()
-                present_labels_sorted = (labels_sorted[nan_mask]).tolist()
-                evaluations = evaluations + self._generate_evaluations(
-                    group['metrics'],
-                    parameters,
-                    {'abs': abs_thresh},
-                    None,
-                    predicted_classes,
-                    present_labels_sorted,
-                )
+            evaluations += self._evaluations_for_group(
```

Comment:
Perhaps: evaluations = evaluations + self._evaluations_for_group(...) for consistency with lines 267, 284, & 261.